modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/PaginationBarTag.java

```diff
@@ -265,19 +269,28 @@ protected int processStartTag() throws Exception {
 
 		jspWriter.write("</li>");
 
-		int totalPages = (int)Math.ceil(_totalItems / _activeDelta);
+		int totalPages = (int)Math.ceil(
+			(double)_totalItems / (double)_activeDelta);
 
-		int ellipsisCount = _activePage + _ellipsisBuffer;
+		if (_ellipsisBuffer != 0) {
+			int activeIndex = _activePage - 1;
```

Comment:
It is better to create the variables just before you need to use them, so maybe we can relocate the activeIndex creation to the bottom just before we need to use it